bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/io/RefreshingOAuth2CredentialsInterceptor.java

```diff
@@ -228,28 +232,67 @@ boolean doRefresh() {
     if (!requiresRefresh) {
       return false;
     }
-    HeaderCacheElement cacheElement = null;
+    HeaderCacheElement cacheElement = refreshCredentialsWithRetry(null);
+    synchronized (isRefreshing) {
+      headerCache.set(cacheElement);
+      isRefreshing.set(false);
+      isRefreshing.notifyAll();
+    }
+    return true;
+  }
+
+
+  protected HeaderCacheElement refreshCredentialsWithRetry(BackOff backoff) {
```

Comment:
private HeaderCacheElement refreshCredentialsWithRetry(@Nullable BackOff backoff) {

Can you document this?